internal/server/migrations.go

```diff
@@ -30,9 +31,15 @@ func (a *API) deprecatedRoutes(noAuthnNoOrg *routeGroup) {
 	type SignupEnabledResponse struct {
 		Enabled bool `json:"enabled"`
 	}
-	addDeprecated(a, noAuthnNoOrg, http.MethodGet, "/api/signup",
-		func(c *gin.Context, _ *api.EmptyRequest) (*SignupEnabledResponse, error) {
+
+	add(a, noAuthnNoOrg, http.MethodGet, "/api/signup", route[api.EmptyRequest, *SignupEnabledResponse]{
```

Comment:
Is this endpoint no longer deprecated? Seems like a weird change

Reply:
I removed the `addDeprecated` function because all it did was define the `routeSettings`, and I don't think we can safely assume that all deprecated routes will have the same `routeSettings`. They should all be `omitFromDocs=true`, but for other settings like `txnOptions` and `omitFromTelemetry` I think it may depend on the route. We may want telemetry for the route to see how many people are still using a deprecated one.

The diff here is moving the logic from `addDeprecated` inline so that we can give each deprecated route its own settings.

This diff is in the `deprecatedRoutes` method, so still documented as deprecated.